src/lib/free-queue/test/free-queue-sab.test.js

```diff
@@ -0,0 +1,113 @@
+        });
+
+        it('should initialize channel data correctly', function() {
+            expect(queue.channelData).to.be.an('array');
+            expect(queue.channelData.length).to.equal(channelCount);
+            queue.channelData.forEach(channel => {
+                expect(channel).to.be.an.instanceof(Float32Array);
+                expect(channel.length).to.equal(bufferLength + 1);
+            });
+        });
+    });
+
+    describe('push', function() {
+        it('should push data into the queue', function() {
+            const input = [new Float32Array([1, 2, 3]), new Float32Array([4, 5, 6])];
+            const result = queue.push(input, 3);
+            expect(result).to.be.true;
+            expect(queue.getAvailableSamples()).to.equal(3);
+        });
+
+        it('should fail to push data when buffer is full', function() {
+            const input = [new Float32Array(queue.getBufferLength()), new Float32Array(queue.getBufferLength())];
+            
+            let result = queue.push(input, queue.getBufferLength());
+            console.log("Push result when buffer is full:", result);
+            expect(result).to.be.true;
+    
+            result = queue.push(input, queue.getBufferLength());
+            console.log("Push result when attempting to overfill buffer:", result);
+            expect(result).to.be.false;
+        });
+    });
+
+    describe('pull', function() {
+        it('should pull data from the queue', function() {
+            const input = [new Float32Array([1, 2, 3]), new Float32Array([4, 5, 6])];
+            queue.push(input, 3);
+            const output = [new Float32Array(3), new Float32Array(3)];
+            const result = queue.pull(output, 3);
+            expect(result).to.be.true;
+            expect(output[0]).to.deep.equal(input[0]);
+            expect(output[1]).to.deep.equal(input[1]);
+        });
+
+        it('should fail to pull data when buffer is empty', function() {
+            const output = [new Float32Array(bufferLength), new Float32Array(bufferLength)];
+            const result = queue.pull(output, bufferLength);
+            expect(result).to.be.false;
+        });
+    });
+
+    describe('Utility Methods', function() {
+        it('should correctly get available samples', function() {
+            expect(queue.getAvailableSamples()).to.equal(0);
+            const input = [new Float32Array([1, 2, 3]), new Float32Array([4, 5, 6])];
+            queue.push(input, 3);
+            expect(queue.getAvailableSamples()).to.equal(3);
+        });
+
+        it('should correctly check if frame is available', function() {
+            expect(queue.isFrameAvailable(3)).to.be.false;
+            const input = [new Float32Array([1, 2, 3]), new Float32Array([4, 5, 6])];
+            queue.push(input, 3);
+            expect(queue.isFrameAvailable(3)).to.be.true;
+        });
+    });
+
+    describe('Performance Tests', function() {
```

Comment:
It looks like Mocha will display timing information for "slow" tests.  This isn't very accurate, but may be enough for our first pass.  I recommend removing the console logging and timeout, and rewriting this section to just have two tests that repeatedly push and pull data, one large and one small.

I tried to modify the existing tests and came up with this which seemed to work well; please feel free to use it or modify as you like.  You may have to adjust the number of iterations to get the timing to show up.

```
  describe('Performance Tests', function() {
    it('should handle large data efficiently', function() {
      const input = [new Float32Array(bufferLength), new Float32Array(bufferLength)];
      const output = [new Float32Array(bufferLength), new Float32Array(bufferLength)];
      for (let i = 0; i < 100000; i++) {
        queue.push(input, bufferLength);
        queue.pull(output, bufferLength);
      }
    });

    it('should handle small data efficiently', function() {
      const input = [new Float32Array(1), new Float32Array(1)];
      const output = [new Float32Array(1), new Float32Array(1)];
      for (let i = 0; i < 1000000; i++) {
        queue.push(input, 1);
        queue.pull(output, 1);
      }
    });
  });
```